config/supervisor/webhook/kustomization.yaml

```diff
@@ -0,0 +1,10 @@
+resources:
+- manifests.yaml
+
+namePrefix: supervisor-
```

Comment:
Should we really use a different prefix?

Reply:
I think yes, because with the current structure

- if we don't differ the namespace, kustomize cannot recognize the webhook config for govmomi mode and supervisor mode and report the following error

```
Error: accumulating resources: accumulation err='accumulating resources from './webhook': '/Users/gongz/zhangg/workspace/src/sigs.k8s.io/cluster-api-provider-vsphere/.build/config/supervisor/webhook' must resolve to a file': recursed merging from path '/Users/gongz/zhangg/workspace/src/sigs.k8s.io/cluster-api-provider-vsphere/.build/config/supervisor/webhook': may not add resource with an already registered id: MutatingWebhookConfiguration.v1.admissionregistration.k8s.io/mutating-webhook-configuration.[noNs]
```
- current structure:
supervisor kustomization uses `base` here 
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/config/supervisor/kustomization.yaml#L12

and `base` contains the `webhook` for govmomi.

Reply:
I think we should move the govmomi webhook to the config/webhook folder

In fact let's do slightly more to make it cleaner. Let's also move config/webhook to config/govmomi/webhook

Reply:
We also have to drop `  - ../webhook` from base/kustomization.yaml and move it to default/kustomization.yaml

Reply:
Sounds good to me. This part is much more clear.